drivers/modbus_rtu/include/modbus_rtu.h

```diff
@@ -0,0 +1,173 @@
+  uint16_t *regs; /**< Registers or byte or bits*/
+} modbus_rtu_message_t;
+
+/**
+ * @brief Modbus structure.
+ */
+typedef struct {
+  /**
+   * @brief  UART port.
+   */
+  uart_t uart;
+  /**
+   * @brief  baudrate.
+   */
+  uint32_t baudrate;
+  /**
+   * @brief  RTS pin. 0 - disable.
+   */
+  gpio_t pin_rts;
+  /**
+   * @brief  RTS signals when transmit.
+   */
+  int pin_rts_enable;
+  /**
+   * @brief  ID device. 0 - master, otherwice slave.
+   */
+  uint8_t id;
+  /**
+   * @brief   Before slave begin to response; usec.
+   */
+  uint32_t timeout;
+  /**
+   * @brief   Template buffer for receive message or send. **Must never be changed
+   *          by the user.**
+   * @internal
+   */
+  uint8_t _buffer[MODBUS_RTU_PACKET_SIZE_MAX];
+  /**
+   * @brief   Buffer size. **Must never be changed by the user.**
+   * @internal
+   */
+  uint8_t _size_buffer;
+  /**
+   * @brief   Mutex for block buffer. **Must never be changed by the user.**
+   * @internal
+   */
+  mutex_t _mutex_buffer;
+  /**
+   * @brief   Timeout beetwin dyte. **Must never be changed by the user.**
+   * @internal
+   */
+  uint32_t _rx_timeout;
+  /**
+   * @brief   pointer on current message. **Must never be changed by the user.**
+   * @internal
+   */
+  modbus_rtu_message_t *_msg;
+  /**
+   * @brief   PID thread. **Must never be changed by the user.**
+   * @internal
+   */
+  kernel_pid_t _pid;
+} modbus_rtu_t;
+
+/**
+ * @brief   Initialize the given modbus.
+ *
+ * Initialize fields of modbus and UART, RTS pin.
+ *
+ * @param[in] modbus    pointer modbus 
+ *
+ * @return              0 on success
+ * @return              less 0 on error
+ */
+int modbus_rtu_init(modbus_rtu_t *modbus);
+/**
+ * @brief   Send request to slave.
+ *
+ * Send request to slave, blocking and wait response.
+ *
+ * @param[in] modbus    pointer modbus 
+ * @param[in] message    pointer modbus message
+ *
+ * @return              0 on success
+ * @return              less 0 on error
+ */
+int modbus_rtu_send_request(modbus_rtu_t *modbus, modbus_rtu_message_t *message);
+/**
+ * @brief   Wait request from master.
+ *
+ * Wait request from master, blocking.
+ * Request will be writeт to message.
+ * Use only slave.
+ *
+ * @param[in] modbus    pointer modbus 
+ * @param[in] message   pointer modbus message
+ *
+ * @return              0 on success
+ * @return              less 0 on error
+ */
+int modbus_rtu_poll(modbus_rtu_t *modbus, modbus_rtu_message_t *message);
+/**
+ * @brief   Send response to master.
+ *
+ * Response must be writeт to message.
+ * Use only slave.
+ * id, func, addr, count - must be set correctly, or not change if it been use in modbus_rtu_send_request
+ *
+ * @param[in] modbus    pointer modbus 
+ * @param[in] message   pointer modbus message
+ *
+ * @return              0 on success
+ * @return              less 0 on error
+ */
+int modbus_rtu_send_response(modbus_rtu_t *modbus, modbus_rtu_message_t *message);
+
+/** @} */
```

Comment:
Nitpick: No `\n` character at the end of this line. (You might want to configure your editor to automatically terminate the last line of every source code file.) The CI tools will enforce that white space errors like this are addressed, as they prevent merging otherwise.